Sources/NIO/System.swift

```diff
@@ -625,8 +625,25 @@ internal enum KQueue {
     @inline(never)
     @discardableResult
     public static func kevent(kq: CInt, changelist: UnsafePointer<kevent>?, nchanges: CInt, eventlist: UnsafeMutablePointer<kevent>?, nevents: CInt, timeout: UnsafePointer<Darwin.timespec>?) throws -> CInt {
+        var timeoutClamped = timeout
+        // Note that a nil timeout waits indefinitely
+        if let ts = timeout {
+            assert(ts.pointee.tv_nsec >= 0 || ts.pointee.tv_nsec < 1_000_000_000, "kevent timeout value is malformed (0 <= tv_nsec < 1_000_000_000)")
+            // Validate that the timeout parameter passed to kevent does not exceed the limits of the Darwin kernel
+            // interval timer (note that - whilst unlikely - these values could change)
+            if ts.pointee.tv_sec < 0  || ts.pointee.tv_sec > 100_000_000 {
```

Comment:
A similarly magic number here. As someone who doesn't know the rules of Darwin kernel timeouts, its strange to see a mixture of `100_000_000` and `1_000_000_000`

Reply:
It's a value splunked out of the kernel, I'm not sure it can be anything other than magic. I can put it in a constant and let kernelItervalTImerMaxiumSecs = 100_000_000?